OneSignalExample/Packages/com.onesignal.unity.android/Editor/OneSignalEditorScript_Android.cs

```diff
@@ -26,73 +26,68 @@
  */
 
 using System.IO;
+using System.Linq;
 using Com.OneSignal.Editor;
 using UnityEditor;
+using UnityEngine;
 
 namespace Com.OneSignal.Android.Editor
 {
     [InitializeOnLoad]
     public class OneSignalEditorScriptAndroid
     {
         static readonly string k_AndroidConfigFolder = $"Packages/{ScopeRegistriesConfig.OneSignalScope}.android/Plugins/Android/OneSignalConfig.plugin";
-        static readonly string k_PackageManifestPath = $"Packages/{ScopeRegistriesConfig.OneSignalScope}.android/package.json";
 
-        static OneSignalEditorScriptAndroid()
-        {
-            CreateOneSignalAndroidManifest();
-            PackageManifestSanityCheck();
-        }
+        const string k_Edm4URepoOwner = "googlesamples";
+        const string k_Edm4URepoName = "unity-jar-resolver";
 
-        /// <summary>
-        /// We can't add EDM4UP into package.json before making sure that google scope registry if available for the project
-        /// The method will
-        ///  * Add Google scope registry to the project manifest.json if necessary
-        ///  * Will update or add configured version of `com.google.external-dependency-manager` dependency into package.json
-        /// </summary>
-        static void PackageManifestSanityCheck()
-        {
-            Manifest mainManifest = new Manifest();
-            mainManifest.Fetch();
+        static readonly string[] k_KnownEdm4ULocations = {
+            "Assets/PlayServicesResolver",
+            "Assets/ExternalDependencyManager"
+        };
 
-            var manifestUpdated = false;
+        static OneSignalEditorScriptAndroid() {
+            CreateOneSignalAndroidManifest();
+            InstallEdm4U();
+        }
 
-            if (!mainManifest.IsRegistryPresent(ScopeRegistriesConfig.GoogleScopeRegistryUrl))
-            {
-                mainManifest.AddScopeRegistry(ScopeRegistriesConfig.GoogleScopeRegistry);
-                manifestUpdated = true;
+        static void InstallEdm4U() {
+            // If Edm4U is already installed we would do nothing.
+            if (k_KnownEdm4ULocations.Any(AssetDatabase.IsValidFolder)) {
```

Comment:
Some developers will move the folder so I think an exists check on an exact path should be a fallback instead of the primary way.

Some ideas I have to detect if EDM4U is in the project.
* Check for it's assembly definition, if it has one
* Check if one of EDM4U's public classes are available. Should be something EDM4U designed to be a public API / class so it won't break between version to version.
* Check the EM4U folder's existence based on it's .meta GUID using `AssetDatabase.GUIDToAssetPath()`. We may want to check for a dll or another file of EDM4U in case the dev forgets to move the .meta file if they move the folder.

Reply:
I've updated the code to check for EDM4U installation. I think this should be good enough:

```charp
static bool IsEdm4UInstalled() {
    var precompiledAssemblies = CompilationPipeline.GetPrecompiledAssemblyNames();
    foreach (var assemblyName in precompiledAssemblies) {
        if (assemblyName.StartsWith("Google.JarResolver")) {
            return true;
        }
    }

    return false;
}
```

Reply:
Ya, that will work no matter where the developer decides to move EDM4U in their project.